src/test/kotlin/org/rustSlowTests/cargo/runconfig/test/CargoTestRunnerTest.kt

```diff
@@ -770,6 +835,22 @@ class CargoTestRunnerTest : CargoTestRunnerTestBase() {
         assertEquals(expectedFormattedTestTree.trimIndent(), getFormattedTestTree(root))
     }
 
+    private fun checkTestLocation(testName: String, fileWithCaret: String) {
+        myFixture.configureFromTempProjectFile(fileWithCaret)
+        val oldCaret = myFixture.caretOffset
+
+        val sourceElement = cargoProjectDirectory.toPsiDirectory(project)!!
+        val configuration = createTestRunConfigurationFromContext(PsiLocation.fromPsiElement(sourceElement))
+        val root = executeAndGetTestRoot(configuration)
+        val test = root.findTestByName(testName)
+        val location = test.getLocation(project, GlobalSearchScope.allScope(project))
+        val desc = location?.openFileDescriptor!!
+        desc.navigateIn(editor)
+
+        val newCaret = myFixture.caretOffset
+        assertEquals(oldCaret, newCaret)
+    }
```

Comment:
The main problem of the current tests that you setup expected state to editor at the beginning.
So, if you drop `desc.navigateIn(editor)` line, all tests won't fail.
Also, we still compare raw offsets and if a test fails, the comparison result won't be human readable
I may suggest something like

```suggestion
    private fun checkTestLocation(testName: String, testProject: TestProject) {
        val expectedText = (testProject.psiFile(testProject.fileWithCaret) as PsiFile).text

        myFixture.configureFromTempProjectFile(testProject.fileWithCaret)
        myFixture.editor.caretModel.moveToOffset(0)

        val sourceElement = cargoProjectDirectory.toPsiDirectory(project)!!
        val configuration = createTestRunConfigurationFromContext(PsiLocation.fromPsiElement(sourceElement))
        val root = executeAndGetTestRoot(configuration)
        val test = root.findTestByName(testName)
        val location = test.getLocation(project, GlobalSearchScope.allScope(project))
        val desc = location?.openFileDescriptor!!
        desc.navigateIn(editor)

        myFixture.checkResult(expectedText)
    }
```